index.html

```diff
@@ -3013,49 +3033,100 @@ <h3>Add Proof Set/Chain</h3>
+a <a>secured data document</a> by verifying its digital proof. The algorithm takes as input:
+        </p>
+
+        <ul>
+          <li>
+|mediaType|, a [=MIME type=]
+          </li>
+          <li>
+|documentBytes|, a [=byte sequence=] whose media type is |mediaType|
+          </li>
+          <li>
+|expectedProofPurpose|, an optional [=string=], used to ensure that the
+<var>proof</var> was generated by the proof creator for the expected reason by the verifier. See
+[[[#proof-purposes]]] for common values.
+          </li>
+          <li>
+|domain|, an optional [=set=] of [=strings=], used by the proof creator to lock a proof to a
+particular security domain, and used by the verifier to ensure that a proof is not being used across
+different security domains
+          </li>
+          <li>
+|challenge|, an optional [=string=] [=challenge=], used by the verifier to ensure that an attacker
+is not replaying previously created proofs
+          </li>
+        </ul>
+
+        <p>
+This algorithm returns a <dfn>verification result</dfn>, a [=struct=] whose [=struct/items=] are:
+        </p>
+        <dl>
+          <dt><dfn data-dfn-for="verification result">status</dfn></dt>
+          <dd>"valid" or "invalid"</dd>
+          <dt><dfn data-dfn-for="verification result">document</dfn></dt>
+          <dd>null if [=verification result/status=] is "invalid" or an [=unsecured data
+          document=]</dd>
+          <dt><dfn data-dfn-for="verification result">media type</dfn></dt>
+          <dd>null if [=verification result/status=] is "invalid" or a [=MIME type=], which may
+          include [=MIME type/parameters=]</dd>
+          <dt><dfn data-dfn-for="verification result" class="lint-ignore">warnings</dfn></dt>
+          <dd>a [=list=] of <dfn data-cite="vc-data-model#dfn-problemdetails">ProblemDetails</dfn>,
+          defaulting to an empty [=list=]</dd>
+          <dt><dfn data-dfn-for="verification result">errors</dfn></dt>
+          <dd>a [=list=] of [=ProblemDetails=], defaulting to an empty [=list=]</dd>
+        </dl>
+
+        <p>
+When a step says "an error MUST be raised", it means to return a [=verification result=] with a
+[=verification result/status=] of "invalid" and a non-empty [=verification result/errors=] list.
         </p>
 
         <ol class="algorithm">
           <li>
-Let <var>proof</var> be set to <var>securedDocument</var>.<var>proof</var>.
+[=Assert=]: the [=MIME type/essence=] of |mediaType| is "`application/vc+ld+json`" or
+"`application/vp+ld+json`".
           </li>
           <li>
-If the <var>proof</var>.<var>type</var>,
-<var>proof</var>.<var>verificationMethod</var>, or
-<var>proof</var>.<var>proofPurpose</var> values are not set,
+Let |securedDocument:map| be the result of running [=parse JSON bytes to an Infra value=] on
+|documentBytes|.
+          </li>
+          <li>
+If |securedDocument| is not a [=map=] or |securedDocument|["`proof`"] is not a [=map=], an error
```

Comment:
Should `|securedDocument|["`proof`"]` have a period separator, as `|securedDocument|.["`proof`"]`? 

There are about 9 similar, seemingly-missing periods, including this one.

Reply:
It shouldn't have a period. This is the syntax demonstrated at https://infra.spec.whatwg.org/#example-map-get, which uses `the_map["key"]` instead of `the_map.key`. Even though Infra uniformly uses `["key"]`, I'm not tied to that syntax and would be equally happy with `the_map.key`, just not the combination of `the_map.["key"]`.

I do think it's important (although not FO important) that the "key" be a literal and ideally a [=link=] to the definition of that key, rather than the `.<var>key</var>` that had been used in various places. Using `<var>` like that prevents you from taking advantage of https://respec.org/docs/#no-unused-vars.